### DISCUSSION TITLE
Start S3 server on the same port as web app

### DISCUSSION THREAD

**Discussion by danielklecha** [jchristn/S3Server#6] [Q&A]
I want to create web app in .NET 8 and listen for requests on port 443.
According to examples instance of S3Server is using different port (8000).
Is it possibe to use the same port?
Is it possible to not use Watson server and pass requests from web app to S3Server library?

Thanks for help!

**Comment by jchristn** [comment]
Hi @danielklecha you certainly could; if you clone from source you could expose the webserver (and thus the routing groups), and attach your routes there.  S3Server primarily relies on the default route though, so you'd be limited to using only the supplied routing groups.

**Comment by jchristn** [comment]
Just merged a PR from a contributor, and added a `.Webserver` property to the `S3Server` object, so this should be easier now.
